docs/reference/data-model.md

```diff
@@ -1,3 +1,373 @@
 
 !!! warning "Work In Progress"
-    This page is currently under construction. Please check back later for updates.
\ No newline at end of file
+    [//]: # (TODO)
+    This page is currently under construction, and not **all** tables in the data model
+    are currently listed on this page. 
+
+### Geneset Table
+The geneset table is the heart of the geneweaver data model. The geneset table is 
+contained in the `production` schema. The geneset table contains the following columns:
+```mermaid
+erDiagram
+    geneset {
+        bigint gs_id
+        integer usr_id
+        bigint file_id
+        varchar gs_name
+        varchar gs_abbreviation
+        integer pub_id
+        integer res_id
+        integer cur_id
+        varchar gs_description
+        integer sp_id
+        integer gs_count
+        integer gs_threshold_type
+        varchar gs_threshold
+        varchar gs_groups
+        varchar gs_attribution_old
+        varchar gs_uri
+        integer gs_gene_id_type
+        date gs_created
+        varchar admin_flag
+        timestamp gs_updated
+        varchar gs_status
+        varchar gsv_qual
+        integer gs_attribution
+        boolean gs_is_edgelist
+    }
+```
+##### Genes & Geneset Values
+Geneset values are the genomic features in the geneset. The geneset values table is
+contained in the `extsrc` schema. The geneset values table is an associative table
+between the geneset table and the gene table. The geneset values table and the genes
+table contain the following columns:
+```mermaid
+erDiagram
+    geneset }o--|| geneset_value : hasA
+    gene }o--|| geneset_value : hasA
+    geneset_value {
+        bigint gs_id
+        bigint ode_gene_id
+        numeric gsv_value
+        bigint gsv_hits
+        character_varying[] gsv_source_list
+        numeric[] gsv_values_list
+        boolean gsv_in_threshold
+        date gsv_date
+    }
+    gene {
+        bigint ode_gene_id
+        varchar ode_ref_id
+        integer gdb_id
+        integer sp_id
+        boolean ode_pref
+        date ode_date
+        bigint[] old_ode_gene_ids
+    }
+```
+
+
+##### Files
+Genesets are created from user uploaded files. The file table is contained in the
+`production` schema. The file table contains the following columns:
+```mermaid
+erDiagram
+    geneset ||--o{  file : createdFrom
+    file {
+        bigint file_id
+        bigint file_size
+        varchar file_uri
+        text file_contents
+        varchar file_comments
+        date file_created
+        text file_changes
+    }
+```
+
+##### Species
+Genesets are of a species, that is, the genomic features in the geneset are from a
```

Comment:
Should you just say "The genomic features in a geneset are from a specific species" and skip the first part of the sentence? This seems a little pedantic, or at least kind of confusing.

Reply:
This is unfortunately the page where I have to be pedantic, but I do think we need to work this better.